packages/labs/src/tag-input/tagInput.tsx

```diff
@@ -167,8 +170,10 @@ export class TagInput extends AbstractComponent<ITagInputProps, ITagInputState>
         const { selectionEnd, value } = event.currentTarget;
         if (event.which === Keys.ENTER && value.length > 0) {
             // enter key on non-empty string invokes onAdd
-            this.setState({ inputValue: "" });
-            Utils.safeInvoke(this.props.onAdd, value);
+            const shouldClearInput = Utils.safeInvoke(this.props.onAdd, value);
+            if (shouldClearInput) {
```

Comment:
I'm iffy on adding a "unique values" switch, because it's doesn't correspond out-of-the-box behavior. Would love to figure out a good way to communicate how to do that in a future PR though. Maybe we could just include a code snippet similar to `NumericInput`s section on uncontrolled usage:

http://blueprintjs.com/docs/#core/components/forms/numeric-input.uncontrolled-mode